doc/apiary/v2/fiware-ngsiv2-reference.apib

```diff
@@ -37,6 +37,105 @@ It is recommended to follow the “url” link values,
+
+* `id` and `type` are the entity ID and entity type, respectively. In the case the entity has an attribute named
+  `id` and/or `type` then they are ignored. For entities without type (i.e. type equals to "" internally), the
+  `type` field is not included
+* All other first level fields correspond to the names of the attributes of the entity.
+  * If the attribute type allows "compact form" and the attribute doesn't have any metadata element, the value of the
+    attribute field is the attribute value itself.
+  * If the attribute type doesn't allow "compact form" and/or the attribute have at least a metadata element, the value
+    of the attrigue field is JSON object including the following fields: `value` (always), `type` (except if the
+    type is empty) and other fields, each one corresponding to the name of one of the metadatas of the attribute.
+* Metadata fields follows the same representation rules that applies to attributes fields (except that a metadata
+  cannot have metadata elements associated).
+
+Canonical format description (TBD).
+
+### Special attribute types
+
+Types are case-insensitive, e.g. `number` is considered an special type, but `Number` is considered a user-defined
+type.
+
+* `number`:  identifies attribute meaning numbers, either integer or float. They are associated to JSON number type.
+They can be used in greater than, lesser than, greater or equal, lesser or equal and range filters. They support compact form:
+
+```
+"temperature": 32.5
+```
+
+* `string`, which represents text string. They are associated to JSON string type. They support compact form:
+
+```
+"msg": "Hello world"
+```
+
+* `boolean`, which represent a logical value (either true or false). They are associated to JSON boolean type.
+They support compact form:
+
+```
+"active": true
+```
+
+* `date`:  identifies attribute meaning a date, in ISO8601 format. They are associated to JSON string type. They
+can be used in greater than, lesser than, greater or equal, lesser or equal and range filters.
+
+
+* `geo:point`: identifies an attribute meaning a geolocation. They are associated to JSON string type. An
+entity can only have one `geo:point` attribute.
+
+```
+{
+  ...
+  "location": {
+    "value": "41.3763726, 2.1864475,14",
+    "type": "geo:point"
+  }
+}
+```
+
+All attribute types not included in the above list are considered user-defined types. User defined types are
+associated to JSON string type.
+
+At creation or updating time, the type has to be coherent with the value of the attribute, e.g the following
+will result in an error response:
+
+```
+{
+  "value": "23.5", -> value should be: 23.5
+  "type": "number"
+}
+
+
+{
+  "value": 23.5,   -> value should be: "23.5"
+  "type": "string"
+}
+
+{
+  "value": "23.5", -> value should be: "23.5" (user defined types always use JSON strings as value)
+  "type": "myType"
+}
+
+{
+  "value": "foo, bar", -> it is not correctly formated as coordinates
+  "type": "geo:point"
+}
+```
```

Comment:
I think examples help. Better to be "verbose" in a specification document that the opossite.